File: Calc.py
import sys
from antlr4 import *
from LabeledExprLexer import LabeledExprLexer
from LabeledExprParser import LabeledExprParser
from EvalVisitor import EvalVisitor


# Programa principal para ejecutar la calculadora en Python
def main(argv):
input_stream = FileStream(argv[1]) if len(argv) > 1 else InputStream(sys.stdin.read())


lexer = LabeledExprLexer(input_stream)
stream = CommonTokenStream(lexer)
parser = LabeledExprParser(stream)
tree = parser.prog()


visitor = EvalVisitor()
visitor.visit(tree)


if __name__ == '__main__':
main(sys.argv)

File: Calc.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.IOException;


// Programa principal para ejecutar la calculadora en Java
public class Calc {
public static void main(String[] args) throws IOException {
// Leer de archivo si se pasa argumento, si no desde consola
CharStream input = CharStreams.fromStream(System.in);


// Lexer y Parser de ANTLR
LabeledExprLexer lexer = new LabeledExprLexer(input);
CommonTokenStream tokens = new CommonTokenStream(lexer);
LabeledExprParser parser = new LabeledExprParser(tokens);


ParseTree tree = parser.prog();


// Visitor para evaluar expresiones
EvalVisitor eval = new EvalVisitor();
eval.visit(tree);
}
}
